code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/BasicPushTemplate.kt

```diff
@@ -56,29 +57,26 @@ internal class BasicPushTemplate : AEPPushTemplate {
              */
             fun getActionButtonFromJSONObject(jsonObject: JSONObject): ActionButton? {
                 return try {
-                    val label = jsonObject.getString(PushTemplateConstants.ActionButtons.LABEL)
+                    val label = jsonObject.getString(ActionButtons.LABEL)
                     if (label.isEmpty()) {
                         Log.debug(
-                            PushTemplateConstants.LOG_TAG,
-                            SELF_TAG, "Label is empty"
+                            LOG_TAG, SELF_TAG, "Label is empty"
                         )
                         return null
                     }
                     var uri: String? = null
-                    val type = jsonObject.getString(PushTemplateConstants.ActionButtons.TYPE)
-                    if (type == PushTemplateConstants.ActionType.WEBURL.name || type == PushTemplateConstants.ActionType.DEEPLINK.name) {
-                        uri = jsonObject.optString(PushTemplateConstants.ActionButtons.URI)
+                    val type = jsonObject.getString(ActionButtons.TYPE)
+                    if (type == ActionType.WEBURL.name || type == ActionType.DEEPLINK.name) {
+                        uri = jsonObject.optString(ActionButtons.URI)
```

Comment:
Good question @navratan-soni 
Looking at the CampaignClassic side implementation when the Action URI is null, the extension falls back to opening the application instead of opening the URI. That seems like a fair alternate to do.. To me it still seems fine to create an action button if the URI is null or not valid